frontend/src/lib/components/Badges/BadgeButton.svelte

```diff
@@ -1,13 +1,17 @@
 <script lang="ts">
-	import { Badge } from ".";
+	import Badge from './Badge.svelte';
 
-	export let type = '';
+    // Add more as necessary. Should be as limited as possible to maximize consistency.
+    type BadgeButtonType = 'badge-success';
+	export let type: BadgeButtonType = 'badge-success';
 	export let icon = '';
 	export let disabled = false;
+
+    $: _type = type as any;
 </script>
 
 <button on:click {disabled} class="button-badge">
-    <Badge {type} {icon}>
-        <slot />
-    </Badge>
+	<Badge type={_type} {icon}>
```

Comment:
could you just do `type={type as any}`? rather then creating a reactive property on line 10?

Reply:
Svelte templates seem to only allow for plain JS.
![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/606ad14a-8218-453c-9792-9a950404ffaf)

I've had the same problem trying to do this: `user?.id`.

Reply:
boo, weird because `user?.id` is valid js these days.